test/forms/TextFormProviderSpec.scala

```diff
@@ -32,13 +33,16 @@ class TextFormProviderSpec extends FieldBehaviours {
       "invalid"
     )
 
-    val invalidTextGen = nonEmptyString.suchThat(!_.matches(formProvider.textAreaRegex))
+    val invalidTextGen = asciiStr.suchThat(_.nonEmpty).suchThat(!_.matches(formProvider.textAreaRegex))
 
     behave.like(fieldThatBindsValidData(form, "value", nonEmptyAlphaString))
     behave.like(mandatoryField(form, "value", "required"))
     behave.like(invalidField(form, "value", "invalid", invalidTextGen))
     behave.like(textTooLongField(form, "value", "tooLong", formProvider.textAreaMaxLength))
-    behave.like((form, "value", "invalid", invalidTextGen))
+
+    "allow punctuation" - {
+      behave.like(fieldThatBindsValidData(form, "value", "Hi, I'm a test on date 10-12-2010 with email test@email.com"))
```

Comment:
Could you add some new lines in here to cover that and also have you considered tabs?